File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        int n=intervals.size();
        vector<vector<int>>ans;
        sort(intervals.begin(),intervals.end());
        ans.push_back(intervals[0]);
        for(int i=1;i<n;i++){
            int currst=intervals[i][0];
            int currend=intervals[i][1];
            if(currst>ans.back()[1]){
                ans.push_back(intervals[i]);
                
            }else{
                ans.back()[1]=max(ans.back()[1],currend);
            }
        }
        return ans;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        List<int[]> ans=new ArrayList<>();
        int n=intervals.length;
        if(n<=1){
            return intervals;
        }

        Arrays.sort(intervals,(a,b)->a[0]-b[0]);
        
        int[] curr=intervals[0];
        for(int i=1;i<n;i++){
            if(intervals[i][0]<=curr[1]){
                curr[1]=Math.max(curr[1],intervals[i][1]);
            }else{
                ans.add(curr);
                curr=intervals[i];
            }
        }
        ans.add(curr);

        return ans.toArray(new int[ans.size()][]);
    }
}